docs/ai/tutorials/snippets/evaluate-with-reporting/WordCountEvaluator.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Text.RegularExpressions;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.AI.Evaluation;
+
+namespace TestAIWithReporting;
+
+public class WordCountEvaluator : IEvaluator
+{
+    public const string WordCountMetricName = "Words";
+
+    public IReadOnlyCollection<string> EvaluationMetricNames => [WordCountMetricName];
+
+    /// <summary>
+    /// Counts the number of words in the supplied string.
+    /// </summary>
+    private static int CountWords(string? input)
+    {
+        if (string.IsNullOrWhiteSpace(input))
+        {
+            return 0;
+        }
+
+        MatchCollection matches = Regex.Matches(input, @"\b\w+\b");
+        return matches.Count;
+    }
+
+    /// <summary>
+    /// Provides a default interpretation for the supplied <paramref name="metric"/>.
+    /// </summary>
+    private static void Interpret(NumericMetric metric)
+    {
+        if (metric.Value is null)
+        {
+            metric.Interpretation =
+                new EvaluationMetricInterpretation(
+                    EvaluationRating.Unknown,
+                    failed: true,
+                    reason: "Failed to calculate word count for the response.");
+        }
+        else
+        {
+            string reason = $"The response is {metric.Value} words long.";
+            if (metric.Value <= 100 && metric.Value > 0)
+                metric.Interpretation = new EvaluationMetricInterpretation(EvaluationRating.Good, reason: reason);
+            else
+                metric.Interpretation = new EvaluationMetricInterpretation(EvaluationRating.Unacceptable, failed: true, reason);
+        }
+    }
+
+    public ValueTask<EvaluationResult> EvaluateAsync(
+        IEnumerable<ChatMessage> messages,
+        ChatResponse modelResponse,
+        ChatConfiguration? chatConfiguration = null,
+        IEnumerable<EvaluationContext>? additionalContext = null,
+        CancellationToken cancellationToken = default)
+    {
+        // Count the number of words in the supplied <see cref="modelResponse"/>.
+        int wordCount = CountWords(modelResponse.Text);
+
+        // Create a <see cref="NumericMetric"/> with value set to the word count.
+        var metric = new NumericMetric(WordCountMetricName, value: wordCount);
+
+        // Attach a default <see cref="EvaluationMetricInterpretation"/> for the metric.
```

Comment:
In the latest version of the samples from the ai-samples repo, there is also a new `metric.Reason` property that is set over here. It is optional so it should be fine to omit it - but I just wanted to mention this in case you wanted to keep this aligned with the sample.